### ISSUE TITLE
应用场景与设定变更时，该如何保证不可学效果

### ISSUE THREAD

**Issue by iberxilong** [HanxunH/Unlearnable-Examples#22]
作者您好，首先十分感谢您的论文和代码贡献，这为我的科研提供了相当大的帮助。
However,目前我们的应用场景是**训练集 和 测试集 的设定要求一致**。例如 训练集随机 5%加class-wise噪声，则测试集也得随机5%加class-wise噪声；
我想请教一下作者orz，关于在上述场景下，要如何添加 保护性噪声 还能 具有 不可学效果 呢？
（任何想法或建议都将是对我极大的帮助QAQ）

**Comment by iberxilong** [comment]
Hello authors, first of all, thank you very much for your paper and code contributions, which have been of considerable help to my research.
However, at present, our application scenario is that the training set and the test set are set to the same requirements. For example, if the training set is randomized by 5% and class-wise noise is added, then the test set must also be randomly 5% with class-wise noise.
I'd like to ask the author Orz about how to add protective noise and still have an unlearnable effect in the above scenario.
(Any ideas or suggestions will be of great help to me QAQ)

**Comment by HanxunH** [comment]
Hi,

这是个好问题。我认为这个场景下还是算不可学习的，因为模型学到的是噪声而不是实际数据里的内容。对于测试来说的话，加了class-wise噪声的数据应当被认成对应的class。

**Comment by iberxilong** [comment]
作者您好，近日我做了些实验，我将 在PoisonCIFAR10上训练好的类噪声，以不同的随机的方式加入到照片中（原本是根据图片的类别来添加噪声），以此来生成训练集P1 和 测试集P2  。 并且**我期望模型在P1上训练，学习，在P2上测试会有较差的效果**。
但实际的实验结果是，在10个epochs左右，模型在训练集和测试集上的准确率都已经超过80%，且随后持续稳步上升。
在上述的加噪声方式基础上，我尝试修改了加噪声的比例，但还是没能做到 预期的效果。（参数如下）
![微信图片_20240229111915](https://github.com/HanxunH/Unlearnable-Examples/assets/104249458/273d78bf-1703-4cd2-8870-cdb5a0f6a2dd)

想请教下作者，我们如何调整，才有可能做到实现我们的期望结果呢？
（任何想法或建议都将是对我极大的帮助QAQ）

**Comment by HanxunH** [comment]
> 作者您好，近日我做了些实验，我将 在PoisonCIFAR10上训练好的类噪声，以不同的随机的方式加入到照片中（原本是根据图片的类别来添加噪声），以此来生成训练集P1 和 测试集P2 。 并且**我期望模型在P1上训练，学习，在P2上测试会有较差的效果**。 但实际的实验结果是，在10个epochs左右，模型在训练集和测试集上的准确率都已经超过80%，且随后持续稳步上升。 在上述的加噪声方式基础上，我尝试修改了加噪声的比例，但还是没能做到 预期的效果。（参数如下） ![微信图片_20240229111915](https://private-user-images.githubusercontent.com/104249458/308783739-273d78bf-1703-4cd2-8870-cdb5a0f6a2dd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0NjUyNTgsIm5iZiI6MTcxMDQ2NDk1OCwicGF0aCI6Ii8xMDQyNDk0NTgvMzA4NzgzNzM5LTI3M2Q3OGJmLTE3MDMtNGNkMi04ODcwLWNkYjVhMGY2YTJkZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxNVQwMTA5MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMDM0ZjJhYjMxMTU3NTE0MDI2ZWM3MTcwYTI0ZmU0ZWNkMWJkYTc2OTMwMDA2NzEwNzUzNTJkMGNmYTJlNGFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NyMTaN4-6xOkMVtDSaB0KUQB-baJsWjdJMshTTfKevM)
> 
> 想请教下作者，我们如何调整，才有可能做到实现我们的期望结果呢？ （任何想法或建议都将是对我极大的帮助QAQ）

如果是class-wise噪声，随机加入到数据集会导致效果不如预期，可以尝试sample-wise的噪声。

**Comment by jedapaw** [comment]
> Dear author, first of all, thank you very much for your paper and code contribution, which has provided considerable help for my scientific research. However, our current application scenario requires that ** the training set and the test set have the same setting requirements ** . For example, if the training set has 5% random class-wise noise, then the test set must also have 5% random class-wise noise. I would like to ask the author orz, in the above scenario, how to add protective noise and still have an unlearnable effect? (Any ideas or suggestions would be of great help to me QAQ)

Greetings,

I am seeking guidance on how to run this model and verify the results. As a beginner, I am finding it challenging to understand the steps involved in executing these models. Can you write here step by step how to set up this and run this? Your assistance would be greatly appreciated.
